packages/gatsby/src/utils/handle-flags.ts

```diff
@@ -0,0 +1,89 @@
+import _ from "lodash"
+import { isCI } from "gatsby-core-utils"
+import realTerminalLink from "terminal-link"
+import { IFlag } from "./flags"
+
+const terminalLink = (text, url): string => {
+  if (process.env.NODE_ENV === `test`) {
+    return `${text} (${url})`
+  } else {
+    return realTerminalLink(text, url)
+  }
+}
+
+const handleFlags = (
+  flags: Array<IFlag>,
+  configFlags: Array<Array<string>>,
+  executingCommand = process.env.gatsby_executing_command
+): { validConfigFlags: Array<IFlag>; message: string } => {
+  // Prepare config flags.
+  // Filter out any flags that are set to false.
+  const filteredConfigFlags = _.toPairs(configFlags).filter(pair => pair[1])
+
+  //  validate flags against current ones
+  //  - any that don't exist — remove silently
+  let validConfigFlags = flags.filter(ae =>
+    filteredConfigFlags.some(flagPair => flagPair[0] === ae.name)
+  )
+
+  // If we're in CI, filter out any flags that don't want to be enabled in CI
+  if (isCI() || process.env.NODE_ENV === `test`) {
```

Comment:
Do we need NODE_ENV check? Don't we want to have this tested too?

Reply:
it was this or mocking isCI — otherwise you couldn't get the same results locally or on CircleCI

Reply:
Is there a pattern for handling this sort of scenario?

Reply:
Mocking probably make more test scenarios possible, because we can have test scenario when CI is true and separate when false.

If we don't have test scenarios like that yet, this is fine but eventually we should really not have checks whether NODE_ENV is `test` in source code, because that make tests not execute some code we would execute normally

Reply:
Cool, I'll change it to mock